src/core/meshcop/leader.hpp

```diff
@@ -26,11 +26,131 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#if OPENTHREAD_MTD
-#include "leader_mtd.hpp"
-#elif OPENTHREAD_FTD
-#include "leader_ftd.hpp"
-#else
-#error "Please define OPENTHREAD_MTD=1 or OPENTHREAD_FTD=1"
-#endif
+/**
+ * @file
+ *   This file includes definitions for a MeshCoP Leader.
+ */
+
+#ifndef MESHCOP_LEADER_HPP_
+#define MESHCOP_LEADER_HPP_
+
+#include <coap/coap_client.hpp>
+#include <coap/coap_server.hpp>
+#include <common/timer.hpp>
+#include <meshcop/tlvs.hpp>
+#include <net/udp6.hpp>
+#include <thread/mle.hpp>
+
+namespace Thread {
+namespace MeshCoP {
+
+OT_TOOL_PACKED_BEGIN
+class CommissioningData
+{
+public:
+    uint8_t GetLength(void) const {
```

Comment:
I mention this because an earlier theme I saw was moving getter/setter operations into the C++ headers.

Thus if this member function has the potential to be used more then once, then this should be a a CPP file not the header file. My point is this, using a get function as an example, is about 2 opcodes. Often it is a single opcode because a register has been previously allocated to the "this" pointer

```
    ldr  r0,_this_pointer
    str  r1,[r0,#member_offset]
```
That is almost the same cost as a subroutine call. The advantage is the compiler can optimize register allocation across that inlined getter/setter function.

This member function has potentially 3 subroutine calls.

These little things add up...  Death by many paper cuts.